src/snap.py

```diff
@@ -211,14 +222,16 @@ def update_mysql_router_exporter_service(
                     "mysqlrouter-exporter.user": config.username,
                     "mysqlrouter-exporter.password": config.password,
                     "mysqlrouter-exporter.url": config.url,
+                    "mysqlrouter-exporter.service-name": self.unit_name.replace("/", "-"),
                 }
             )
             _snap.start([self._EXPORTER_SERVICE_NAME], enable=True)
         else:
+            _snap.stop([self._EXPORTER_SERVICE_NAME], disable=True)
```

Comment:
not to my understanding, because the set variable is only used in the snap script when starting mysql-router-exporter. if the variable is unset while the service is still running, the service will be unaffected as the args are passed in-line when the script is invoked [here](https://github.com/canonical/charmed-mysql-snap/blob/8.0/edge/snap/local/start-mysql-router.sh#L7-L16)